### ISSUE TITLE
`@types/resize-observer-browser` conflicts with `resize-observer-polyfill`

### ISSUE THREAD

**Issue by bennypowers** [material-components/material-components-web-components#2047]
**Describe the bug**
Installing `@material/mwc-linear-progress` alongside `resize-observer-polyfill` causes typescript errors

<details>
<summary>
Duplicate identifier 'ResizeObserverCallback'.
</summary>

```
node_modules/@types/resize-observer-browser/index.d.ts:35:6 - error TS2300: Duplicate identifier 'ResizeObserverCallback'.

35 type ResizeObserverCallback = (entries: ReadonlyArray<ResizeObserverEntry>, observer: ResizeObserver) => void;
        ~~~~~~~~~~~~~~~~~~~~~~

  node_modules/resize-observer-polyfill/src/index.d.ts:13:15
    13     interface ResizeObserverCallback {
                     ~~~~~~~~~~~~~~~~~~~~~~
    'ResizeObserverCallback' was also declared here.

node_modules/resize-observer-polyfill/src/index.d.ts:13:15 - error TS2300: Duplicate identifier 'ResizeObserverCallback'.

13     interface ResizeObserverCallback {
                 ~~~~~~~~~~~~~~~~~~~~~~

  node_modules/@types/resize-observer-browser/index.d.ts:35:6
    35 type ResizeObserverCallback = (entries: ReadonlyArray<ResizeObserverEntry>, observer: ResizeObserver) => void;
            ~~~~~~~~~~~~~~~~~~~~~~
    'ResizeObserverCallback' was also declared here.

node_modules/resize-observer-polyfill/src/index.d.ts:19:18 - error TS2717: Subsequent property declarations must have the same type.  Property 'contentRect' must be of type 'DOMRectReadOnly', but here has type 'DOMRectReadOnly'.

19         readonly contentRect: DOMRectReadOnly;
                    ~~~~~~~~~~~

  node_modules/@types/resize-observer-browser/index.d.ts:39:14
    39     readonly contentRect: DOMRectReadOnly;
                    ~~~~~~~~~~~
    'contentRect' was also declared here.
```

</details>

**To Reproduce**
1. `npm i -D resize-observer-polyfill@1.5.1`
1. `npm i -S @material/mwc-linear-progress@0.20.0`

**Expected behavior**
No Type errors reported

**Screenshots**

**Browser Version (please complete the following information):**
 - TypeScript 4.1.3

**Additional context**
Suggestions for fixes:
- inline the types
- use the polyfill's types
- use `any` with comments

**Comment by bennypowers** [comment]
<details><summary>my workaround via <code>patch-package</code></summary>

```diff
diff --git a/node_modules/@types/resize-observer-browser/index.d.ts b/node_modules/@types/resize-observer-browser/index.d.ts
index 5b0bcbf..7ee5795 100644
--- a/node_modules/@types/resize-observer-browser/index.d.ts
+++ b/node_modules/@types/resize-observer-browser/index.d.ts
@@ -1,43 +1 @@
-// Type definitions for non-npm package resize-observer-browser 0.1
-// Project: https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver, https://developers.google.com/web/updates/2016/10/resizeobserver, https://wicg.github.io/ResizeObserver/
-// Definitions by: Chives <https://github.com/chivesrs>
-//                 William Furr <https://github.com/wffurr>
-//                 Alexander Shushunov <https://github.com/AlexanderShushunov>
-// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.7
-
-interface Window {
-    ResizeObserver: typeof ResizeObserver;
-}
-
-interface ResizeObserverOptions {
-    /**
-     * Sets which box model the observer will observe changes to. Possible values
-     * are `content-box` (the default), and `border-box`.
-     *
-     * @default 'content-box'
-     */
-    box?: 'content-box' | 'border-box';
-}
-
-interface ResizeObserverSize {
-    readonly inlineSize: number;
-    readonly blockSize: number;
-}
-
-declare class ResizeObserver {
-    constructor(callback: ResizeObserverCallback);
-    disconnect(): void;
-    observe(target: Element, options?: ResizeObserverOptions): void;
-    unobserve(target: Element): void;
-}
-
-type ResizeObserverCallback = (entries: ReadonlyArray<ResizeObserverEntry>, observer: ResizeObserver) => void;
-
-interface ResizeObserverEntry {
-    readonly target: Element;
-    readonly contentRect: DOMRectReadOnly;
-    readonly borderBoxSize?: ReadonlyArray<ResizeObserverSize>;
-    readonly contentBoxSize?: ReadonlyArray<ResizeObserverSize>;
-    readonly devicePixelContentBoxSize?: ReadonlyArray<ResizeObserverSize>;
-}
+export * from 'resize-observer-polyfill';
\ No newline at end of file
```

</details>

**Comment by e111077** [comment]
fixed in #2081 synced in #2091
